### DISCUSSION TITLE
LaTeX documents: figure options passed to \includegraphics[]{}

### DISCUSSION THREAD

**Discussion by DavidJesse21** [quarto-dev/quarto-cli#5175] [Q&A] [answered]
Hey everyone,

when producing LaTeX documents with quarto (e.g. a normal pdf document or beamer slides), the resulting/intermediate .tex file will contain `\includegraphics{example.pdf}` for including plots.
Is there a possibility to specify options that we wish to pass to the `includegraphics[<options>]{example.pdf}` command?
In my particular case, I would like to specify the scale argument like it is done [here](https://www.overleaf.com/learn/latex/Questions/How_do_I_specify_the_size_of_an_image_in_LaTeX%3F).
I am aware of the possibility to specify the `fig-height` and `fig-width` cell options but that does not give me enough flexibility to obtain the results I wish.

Cheers
David

**Comment by cderv** [comment]
There is no Markdown syntax to pass option to `includegraphics` in LaTeX. This is a Pandoc limitations. 
From https://pandoc.org/MANUAL.html#extension-link_attributes
> For HTML and EPUB, all known HTML5 attributes except width and height (but including srcset and sizes) are passed through as is. Unknown attributes are passed through as custom attributes, with data- prepended. **The other writers ignore attributes that are not specifically supported by their output format**.

It seems `scale` is not among options known by Pandoc's LaTeX writer .

What you can do is use raw LaTeX in your document. If you target multi format, you can use [condition content](https://quarto.org/docs/authoring/conditional.html) to insert the raw latex only when output is PDF.

**Reply by DavidJesse21** [reply]
Thank you for your quick reply!
Multiple formats is not an issue for me as my problem, at least in this case, concerns creating beamer slides only.
But can you maybe tell me what your approach would be using raw LaTeX code?
I am concerned with creating plots using ggplot and not with including an existing image.
My initial approach would be to create the plot object and save it, e.g. via `ggsave()`,  include the resulting file and delete it right after that.
Is that how you would do it as well?

**Comment by baptiste** [comment]
(assuming a R/knitr document)

You could ask knitr to produce directly the latex code for the figure and pass the scale value via `out.extra`. Note that `\setkeys{Gin}` in the pandoc preamble also gets in the way, as set values of width and height overtake any scale passed to includegraphics, so one workaround is to resets those ([with `!`, "obviously"](https://tex.stackexchange.com/a/297912/269911)).

````
---
title: "Untitled"
format: 
   pdf:
     keep-tex: true
     keep-md: true
---

\setkeys{Gin}{width=!,totalheight=!}

```{r, dev='pdf'}
library(knitr)
knit_hooks$set(plot = hook_plot_tex)
plot(1,1)
```


```{r, dev='pdf', out.extra='scale=0.5'}
plot(1,1)
```

````

(that being said, I would not recommend this approach – I find it best to set the width/height directly with `width=50%` etc.)

**Comment by cderv** [comment, marked as answer]
FWIW related issue - we will track that there now 

* https://github.com/quarto-dev/quarto-cli/issues/6239
